RestartManager.js

```diff
@@ -26,6 +28,12 @@ const RestartManager = (() => {
 
     function restartApp(onlyIfUpdateIsPending = false) {
         if (_allowed) {
+            if (_inProgress) {
+                log("A restart request is already in progress or queued");
+                return;
+            }
+            // The restart won't execute if `onlyIfUpdateIsPending === true` and there is no pending update.
+            _inProgress = !onlyIfUpdateIsPending || !!(NativeCodePush.getUpdateMetadata(CodePush.UpdateState.PENDING));
```

Comment:
I've been trying all day to get it work like that to, unfortunately, no avail. The general issue is that receiving information from the plugin requires using a promise. Because the promise is not synchronous, the `_inProgress` field takes a non-negligible amount of time to update, and calling `restartApp` twice in a row causes it to crash still because `_inProgress` has not been updated yet. In retrospect, I think this line is incorrect as well because it always sets `_inProgress` to true because `NativeCodePush.getUpdateMetadata` returns a promise (which passes the `!!` check) and not an actual boolean. I wrote a new test case so we now have coverage of a) restarts in progress block other restarts and b) restarts that are ignored do not block additional restarts but unfortunately this seems to be a harder problem than i had thought initially.